aspnet/mvc/overview.rst

```diff
@@ -1,8 +1,288 @@
+
+.. code-block:: text
+
+  <ul>
+    @for (int i = 0; i < 5; i++) {
+      <li>List item @i</li>
+    }
+  </ul>
+
+Using the Razor view engine you can define :doc:`layouts </mvc/views/layout>`, :doc:`partial views </mvc/views/partial>` and replaceable sections.
+
+Strongly typed views
+^^^^^^^^^^^^^^^^^^^^
+
+Razor views in MVC can be strongly typed based on your model. A strongly-typed view explicitly specifies the model type for the view. You can then leverage the compiler and IntelliSense to ensure that references to property names and types in your views are correct:
+
+.. code-block:: html
+
+  @model IEnumerable<Product>
+  <ul> 
+      @foreach (Product p in Model) 
+      { 
+          <li>@p.Name</li>
+      }
+  </ul>
+
+Tag Helpers
+^^^^^^^^^^^
+
+:doc:`Tag Helpers </mvc/views/tag-helpers/intro>` enable you to dynamically generate or modify the HTML returned to the client. You can use tag helpers to define custom tags or to modify the behavior of existing tags. Tag Helpers bind to specific elements based on the element name and its attributes. Tag Helpers give you all of the benefits of server-side rendering while still preserving an HTML editing experience.
+
+For example, you can use the built-in ``LinkTagHelper`` to create a link to the ``Login`` action of your ``AccountsController``:
+
+.. code-block:: html
+  :emphasize-lines: 3
+  
+  <p>
+      Thank you for confirming your email. 
+      Please <a asp-controller="Account" asp-action="Login">Click here to Log in</a>.
+  </p>
+
+Or you can use the ``EnvironmentTagHelper`` to include different scripts in your views (for example, raw or minified) based on the runtime environment, such as Development, Staring, or Production:
+
+.. code-block:: html
+  :emphasize-lines: 1,3-4,9
+
+  <environment names="Development">
+      <script src="~/lib/jquery/dist/jquery.js"></script>
+  </environment>
+  <environment names="Staging,Production">
+      <script src="https://ajax.aspnetcdn.com/ajax/jquery/jquery-2.1.4.min.js"
+              asp-fallback-src="~/lib/jquery/dist/jquery.min.js"
+              asp-fallback-test="window.jQuery">
+      </script>
+  </environment>
+
+ASP.NET Core MVC includes a full set of built-in Tag Helpers that you can use or you can build your own.
+
+View Components
+^^^^^^^^^^^^^^^
+
+:doc:`View Components </mvc/views/view-components>` are like :doc:`partial views </mvc/views/partial>` with associated application buisiness logic. View Components can have asynchronous logic and can be reused throughout your application's views.
```

Comment:
:doc:`View Components </mvc/views/view-components>` allow you to package  rendering logic and use it throughout the application.

@rynowak  - we need the elevator pitch for VC's in this **Overview of MVC** doc. This is for new folks considering ASP.NET Core.   Asynch is a detail, why wouldn't they be async?